guppylang/engine.py

```diff
@@ -107,9 +109,15 @@ def reset(self) -> None:
         """Resets the compilation cache."""
         self.parsed = {}
         self.checked = {}
+        self.compiled = {}
```

Comment:
note to reviewers: this is a drive-by change, as the original omission of a `self.compiled` reset appeared accidental.

Please check whether or not this is the case. I don't want to cause a performance regression if resetting the compiled dictionary is avoided for e.g. caching.